Libraries/src/net/relinc/libraries/staticClasses/SPOperations.java

```diff
@@ -49,7 +53,7 @@
 	public static String strainGaugeImageLocation = "/net/relinc/libraries/images/strainGaugeImage.png";
 	public static String relLogoImageLocation = "/net/relinc/libraries/images/rel-logo.png";
 	public static String surePulseLogoImageLocation = "/net/relinc/libraries/images/SURE-Pulse_DP_Logo.png";
-
+	public static String SAMPLE_TYPE = null;
```

Comment:
?? Not sure what this is for...

Reply:
"Variable used in lambda expression should be final or effectively final"
SPOperations loadSample method has a lambda that for some reason required the string sampleType to be final.

Reply:
Ok, so you can declare 
```java
final String sampleType;
```
Then make the assignment inside the if-else and it should work. 

At any rate having `SAMPLE_TYPE` with that much scope is unnecessary. If you must create another variable, you could just do
```java
final String sampleTypeFinal = sampleType
```
at the lowest scope possible. I have done this before. But I tried this and I don't think it's  necessary here.